### PR TITLE
[Citrix ADC] Add Integration Package with System Data Stream

### PR DESCRIPTION
**Pull request by harnish-crest**
- Enhancement

## What does this PR do?

- Added 1 data stream (System metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/citrix_adc directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #3746
- Relates #4322 
- Relates #4364
- Relates #4418 
- Relates #4473

## Screenshots
citrix_adc-system-dashboard

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM

File: packages/citrix_adc/_dev/build/docs/README.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 The Citrix ADC integration allows you to monitor your Citrix ADC instance. Citrix ADC is an application delivery controller that performs application-specific traffic analysis to intelligently distribute, optimize, and secure Layer 4 - Layer 7 (L4–L7) network traffic for web applications.
 
-Use the Citrix ADC integration to collect metrics related to the interface, lbvserver, and service. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference logs when troubleshooting an issue.
+Use the Citrix ADC integration to collect metrics related to the system. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference logs when troubleshooting an issue.
```
Comment (muthu-mps) [inline packages/citrix_adc/_dev/build/docs/README.md:7]: why are we removing all the other datastreams and adding only system ?

File: packages/citrix_adc/_dev/build/docs/README.md [hunk 2]
```diff
@@ -54,3 +58,11 @@ This is the `service` data stream. With the help of the service endpoint, metric
 {{event "service"}}
 
 {{fields "service"}}
+
+### System
+
+This is the `system` data stream. With the help of the system endpoint, metrics like memory in use, total system memory, CPU count can be collected along with other statistics for system resources. `system` data stream collects metrics related to CPU utilization, disk usage, and memory.
```
Comment (muthu-mps) [inline packages/citrix_adc/_dev/build/docs/README.md:64]: Do you think the second sentence is more of a repeated one ?

File: packages/citrix_adc/data_stream/system/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      field: json.system.starttime
+      target_field: citrix_adc.system.start.time
+      ignore_failure: true
+      formats:
+      - EEE MMM dd HH:mm:ss yyyy
+      - EEE MMM dd HH:mm:ss.SSSSSS yyyy
+      - EEE MMM  d HH:mm:ss yyyy
+      - EEE MMM  d HH:mm:ss.SSSSSS yyyy
+  - script:
+      description: Convert megabyte to byte.
```
Comment (muthu-mps) [inline packages/citrix_adc/data_stream/system/elasticsearch/ingest_pipeline/default.yml:120]: Why do we need to convert the MB to bytes ?

File: packages/citrix_adc/manifest.yml [hunk 4]
```diff
@@ -33,14 +33,18 @@ screenshots:
...
+    type: image/png
 policy_templates:
   - name: Citrix ADC
     title: Citrix ADC metrics
     description: Collect Citrix ADC metrics.
     inputs:
       - type: httpjson
         title: Collect Citrix ADC metrics
-        description: Collect interface, lbvserver, and service metrics from Citrix ADC product.
+        description: Collect interface, lbvserver, service, and system metrics from Citrix ADC product.
```
Comment (muthu-mps) [inline packages/citrix_adc/manifest.yml:47]: nit: please remove the extra comma.

**Comment by harnish-crest** [comment]
> In the dashboards can we remove the time when the application was started. The Kibana system memory shows 16 MB and used memory shows 218MB. Is it representing the data correctly ?

The reason behind providing this visualization is, for e.g If a user identifies that CPU/Memory Utilization keeps on increasing, they'll know that this Netscaler instance has been running since it keeps increasing so they can reboot it or something. So all the necessary information would be available on the dashboards themselves, User won't need to go in discover to identify since what date the Netscaler appliance is configured.

File: packages/citrix_adc/data_stream/system/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (harnish-crest) [reply packages/citrix_adc/data_stream/system/elasticsearch/ingest_pipeline/default.yml:120]: Kibana provides functionality to automatically convert bytes into human-readable units, to provide support for this functionality in the dashboard we have converted MB into bytes.

**Comment by muthu-mps** [comment]
> > In the dashboards can we remove the time when the application was started. The Kibana system memory shows 16 MB and used memory shows 218MB. Is it representing the data correctly ?
> 
> The reason behind providing this visualization is, for e.g If a user identifies that CPU/Memory Utilization keeps on increasing, they'll know that this Netscaler instance has been running since it keeps increasing so they can reboot it or something. So all the necessary information would be available on the dashboards themselves, User won't need to go in discover to identify since what date the Netscaler appliance is configured.

Can you please provide an example where we need to reboot the instance based on utilisation ?

**Comment by harnish-crest** [comment]
> > > In the dashboards can we remove the time when the application was started. The Kibana system memory shows 16 MB and used memory shows 218MB. Is it representing the data correctly ?
> > 
> > 
> > The reason behind providing this visualization is, for e.g If a user identifies that CPU/Memory Utilization keeps on increasing, they'll know that this Netscaler instance has been running since it keeps increasing so they can reboot it or something. So all the necessary information would be available on the dashboards themselves, User won't need to go in discover to identify since what date the Netscaler appliance is configured.
> 
> Can you please provide an example where we need to reboot the instance based on utilisation ?

We don’t have any concrete examples, but this was the thought process we were having while working on this dashboard. If you don’t see any value getting provided to the end user, we can remove it.

**Comment by muthu-mps** [comment]
> > > > In the dashboards can we remove the time when the application was started. The Kibana system memory shows 16 MB and used memory shows 218MB. Is it representing the data correctly ?
> > > 
> > > 
> > > The reason behind providing this visualization is, for e.g If a user identifies that CPU/Memory Utilization keeps on increasing, they'll know that this Netscaler instance has been running since it keeps increasing so they can reboot it or something. So all the necessary information would be available on the dashboards themselves, User won't need to go in discover to identify since what date the Netscaler appliance is configured.
> > 
> > 
> > Can you please provide an example where we need to reboot the instance based on utilisation ?
> 
> We don’t have any concrete examples, but this was the thought process we were having while working on this dashboard. If you don’t see any value getting provided to the end user, we can remove it.

Please remove the visualisation.

File: packages/citrix_adc/_dev/build/docs/README.md (hunk 2 above)
Comment (harnish-crest) [reply packages/citrix_adc/_dev/build/docs/README.md]: Updated, thanks!

File: packages/citrix_adc/_dev/build/docs/README.md (hunk 1 above)
Comment (harnish-crest) [reply packages/citrix_adc/_dev/build/docs/README.md]: updated, thanks!

**Comment by muthu-mps** [comment]
> > I am just wondering why are we converting the MB to bytes which adds extra overhead at the ingest pipeline and then showing the values as MB in dashboards ? Is it not possible to show the MB directly in Kibana without any conversion ?
> 
> As per my understanding, currently, there is no direct way to convert MB to other formats, the current possible approach is to convert Mb to bytes so that the bytes can be automatically converted to human-readable format(KB, MB, TB, ...) using the Kibana visualization itself, let me know if you have any other suggestions on the same.

For now this is fine. But we may need to look for an option to do it directly instead of applying convertions.

**Pull request merged**